### PR TITLE
Adds compatibility for suffix offer identifiers

### PR DESCRIPTION
**Pull request by vegaro**
We ask folks to configure the same promotional offer ids for all products, and this works for products that don't belong to the same subscription group, but promotional offer ids have to be unique per product inside the same subscription group.

This PR adds support for promotional offer ids that can be configured using a suffix that can be set in the dashboard. This way promo offer ids can be set as "product_id_1_rc_refund_offer" and "product_id_2_rc_refund_offer", and the Customer Center will be able to load them and AppStore connect will allow to create them.

### CODE REVIEW

File: RevenueCatUI/CustomerCenter/Data/LoadPromotionalOfferUseCase.swift [hunk 1]
```diff
@@ -49,8 +49,12 @@ class LoadPromotionalOfferUseCase: LoadPromotionalOfferUseCaseType {
...
             }
 
-            guard let discount = subscribedProduct.discounts.first(where: {
-                $0.offerIdentifier == promoOfferDetails.iosOfferId
+            guard let discount = subscribedProduct.discounts.first(where: { discount in
+                guard let offerIdentifier = discount.offerIdentifier else {
+                    return false
+                }
+                return offerIdentifier == promoOfferDetails.iosOfferId ||
+                       offerIdentifier.hasSuffix("_\(promoOfferDetails.iosOfferId)")
```
Comment (tonidero) [inline RevenueCatUI/CustomerCenter/Data/LoadPromotionalOfferUseCase.swift:57]: Should we first search for the exact match and if it's not found, then find the suffix? I'm thinking that with the current implementation we might find a version with the suffix even if we have another one that is an exact match.

**Review by tonidero — COMMENTED** [review]
Just a comment but looking good!

File: RevenueCatUI/CustomerCenter/Data/LoadPromotionalOfferUseCase.swift (hunk 1 above)
Comment (vegaro) [reply RevenueCatUI/CustomerCenter/Data/LoadPromotionalOfferUseCase.swift]: I think that's better yes. I updated it

**Review by tonidero — APPROVED** [review]
LGTM!

**Pull request merged**